program/src/instructions/t22_sell_nft_trade_pool.rs

```diff
@@ -146,7 +146,10 @@ pub struct SellNftTradePoolT22<'info> {
     pub taker_broker: Option<UncheckedAccount<'info>>,
 
     /// The optional cosigner account that must be passed in if the pool has a cosigner.
-    /// Checks are performed in the handler.
+    /// Missing check is performed in the handler.
+    #[account(
+        constraint = cosigner.key() == pool.cosigner @ ErrorCode::BadCosigner,
```

Comment:
Nit: would this be easier to do in a pool constraint? eg `constraint = pool.cosigner == cosigner.map(|c| c.key()) @ ErrorCode::BadCosigner`

Reply:
`pool.cosigner` is a `Pubkey` not a `Option<Pubkey>` so this doesn't work. The other nice thing about doing it on the `cosigner` account is the comparison only gets run if a `cosigner` is passed in since it's an optional account.